databricks/koalas/data_type_ops.py

```diff
@@ -0,0 +1,68 @@
+import datetime
+import decimal
+import numpy as np
+
+from pandas.api.types import CategoricalDtype
+
+
+class DataTypeOps:
+    def __new__(cls, dtype, spark_type):
```

Comment:
Shall we only allow `dtype: Dtype`?

Reply:
Added type annotations. I'm wondering what is `spark_type`'s usage in `__new__`?

Reply:
We can't decide the actual type class without it.
For example, without it, how do you decide the actual class if dtype is `np.dtype('object')`?

Reply:
Got it, thanks!